src/Classes/ItemsTab.lua

```diff
@@ -2598,18 +2596,36 @@ function ItemsTabClass:AddImplicitToDisplayItem()
 				table.sort(modList[i], function(a, b)
 					local modA = a.mod
 					local modB = b.mod
-					for i = 1, m_max(#modA, #modB) do
-						if not modA[i] then
+					if modA.group ~= modB.group then
+						if modA.group:match("PinnaclePresence") then
+							return false
+						elseif modB.group:match("PinnaclePresence") then
 							return true
-						elseif not modB[i] then
+						elseif modA.group:match("UniquePresence") then
 							return false
-						elseif modA.statOrder[i] ~= modB.statOrder[i] then
-							return modA.statOrder[i] < modB.statOrder[i]
+						else
+							return true
+						end
+					end
+					for j = 1, m_max(#modA, #modB) do
+						if not modA[j] then
+							return true
+						elseif not modB[j] then
+							return false
+						elseif modA.statOrder[j] ~= modB.statOrder[j] then
+							return modA.statOrder[j] < modB.statOrder[j]
+						else
+							local modAVal = tonumber(modA[j]:match("[%d%.]+"))
+							local modBVal = tonumber(modB[j]:match("[%d%.]+"))
```

Comment:
```suggestion
							local modAVal = tonumber(a.defaultOrder:match("%d+$"))
							local modBVal = tonumber(b.defaultOrder:match("%d+$"))
```
Some mods are descending values this uses the modId to sort them.
Before
![image](https://user-images.githubusercontent.com/31533893/219829852-ad431993-7ef4-4015-8dfb-395d8d2f660a.png)
After
![image](https://user-images.githubusercontent.com/31533893/219829833-ba192172-278f-48dc-958c-571e53b01a62.png)